packages/webdriver-utils/src/providers/automateProvider.js

```diff
@@ -83,12 +83,23 @@ export default class AutomateProvider extends GenericProvider {
           percyBuildUrl: this.buildInfo.url,
           state: 'begin'
         });
+        // Selenium Hub, set status error Code to 13 if an error is thrown
+        if (result?.status && result.status === 13) throw new Error(result?.value || 'Got invalid error response');
         this._markedPercy = result.success;
         return result;
       } catch (e) {
         log.debug(`[${name}] : Could not mark Automate session as percy`);
         log.error(`[${name}] : error: ${e.toString()}`);
-        return null;
+        /**
+         * ERROR response format from SeleniumHUB `{
+         * sessionId: ...,
+         * status: 13,
+         * value: { error: '', message: ''}
+         * }
+         */
```

Comment:
Add a comment Mentioning reason why we need to handle it at 2 places separately

Reply:
Selenium HUB response is 2 different ways, based on dialect for older version of selenium ( when dialect is OSS) if error is throw from Selenium hub.
it send response as 200 with response body as { sessionId: Number, status: 13, value: String Error Message }

```
result => {
  sessionId: 'af05d6069e305cf9d62ced9dbd0ddfbd12b8a59b',
  status: 13,
  value: 'Error executing browserstack_executor command. [PERCY_UNSUPPORTED_COMBINATION_PASSED] Selenium Version 3.5.2 is not supported on Percy'
}
Error ==> CHECK  Error: Error executing browserstack_executor command. [PERCY_UNSUPPORTED_COMBINATION_PASSED] Selenium Version 3.5.2 is not supported on Percy
```

where for newer version when dialect is W3C
it send response as 500, with response as 
body: JSON STRING contains ( value: { error: '', message: '' }}

```
[percy:core] Percy has started! (1033ms)
[percy] [percy_poc_tile_0] : Starting automate screenshot ...
Error ==> CHECK  Error: 500 Internal Server Error
    at request (/Users/shivamsingh/Documents/percy/cli/packages/sdk-utils/dist/request.js:26:25)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Driver.executeScript (file:///Users/shivamsingh/Documents/percy/cli/packages/webdriver-utils/dist/driver.js:49:34)
    at async AutomateProvider.browserstackExecutor (file:///Users/shivamsingh/Documents/percy/cli/packages/webdriver-utils/dist/providers/automateProvider.js:152:15)
    at async file:///Users/shivamsingh/Documents/percy/cli/packages/webdriver-utils/dist/providers/automateProvider.js:58:22
    at async TimeIt.run (file:///Users/shivamsingh/Documents/percy/cli/packages/webdriver-utils/dist/util/timing.js:6:31)
    at async AutomateProvider.percyScreenshotBegin (file:///Users/shivamsingh/Documents/percy/cli/packages/webdriver-utils/dist/providers/automateProvider.js:56:12)
    at async AutomateProvider.screenshot (file:///Users/shivamsingh/Documents/percy/cli/packages/webdriver-utils/dist/providers/automateProvider.js:32:22)
    at async WebdriverUtils.automateScreenshot (file:///Users/shivamsingh/Documents/percy/cli/packages/webdriver-utils/dist/index.js:23:30)
    at async file:///Users/shivamsingh/Documents/percy/cli/packages/core/dist/api.js:142:18 {
  response: {
    status: 500,
    statusText: 'Internal Server Error',
    headers: {
      'cache-control': 'no-cache',
      'content-type': 'application/json; charset=utf-8',
      date: 'Wed, 25 Oct 2023 13:47:33 GMT',
      'source-from': '02',
      'content-length': '352',
      connection: 'keep-alive'
    },
    body: '{"sessionId":"35d36f6ec086d27a6f8481eca08b6bfae585cdea","status":13,"value":{"error":"Error executing browserstack_executor command. [PERCY_UNSUPPORTED_COMBINATION_PASSED] macsie Sierra is not supported on Percy","message":"Error executing browserstack_executor command. [PERCY_UNSUPPORTED_COMBINATION_PASSED] macsie Sierra is not supported on Percy"}}'
  }
 ```
refer: https://github.com/browserstack/SeleniumHub/blob/master/helpers/customSeleniumHandling/customExecutorHelper.js#L20C4-L20C4

Therefore need to handle in 2 different ways

Reply:
Not blocking on it but do add a comment 1-2 line that we are handling because we are handling multiple kind exception structure in responses.

Reply:
SeleniumHub sends 2 type of response while handling with different version of Selenium.
-> For 1st it sends 200 response but in response a status code 13 is set, that represents an error from Hub.
-> For 2nd it send 500 with response in different way.